### PR TITLE
Introducing possiblity to define the server baseUrl using environment variable

### PR DESCRIPTION
**Pull request by jadelkhoury**

### CODE REVIEW

**Comment by jadelkhoury** [comment]
@berezovskyi it would be appreciated if you can look at the suggested mtl changes? It's mainly java code, so hope it is possible.

**Comment by jamsden** [comment]
baseURL is a pretty fundamental characteristic of a server, and not something that is expected to change very often. Environment variables are intended for more variable parameters.

**Comment by jadelkhoury** [comment]
The property is only read once at server startup. It's the same behaviour as the current approach with ServletContext and/or SystemProperty. 
I believe Environment properties was an easier approach to configure Docker containers.

File: org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl [hunk 1]
```diff
@@ -89,15 +85,17 @@ public class [javaClassNameForAdaptorServletListener(anAdaptorInterface) /] impl
     public void contextInitialized(final ServletContextEvent servletContextEvent)
     {
         //These are default values. You can modify any of them early in this method.
-        String basePathEnvKey = "LYO_BASE";
-        String basePathContextPropertyKey = String.format("%s.%s", [javaClassNameForAdaptorServletListener(anAdaptorInterface) /].class.getPackage().getName(), "baseurl");
+        String basePathKey = "baseurl";
```
Comment (berezovskyi) [inline org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl:88]: We should write in the changelog that `LYO_BASE` will be renamed to `LYO_BASEURL` to maintain consistency of all property setting mechanism.

@jadelkhoury I think this the perfect time to add CHANGELOG.md to LyoD

**Review by berezovskyi — CHANGES_REQUESTED** [review]
See the comments inside. Looks good otherwise!

**Comment by berezovskyi** [comment]
@jamsden, I don't think there is any major concern with this PR:

1. Lyo Designer is not Lyo. Lyo libs are expected to be updated in-place but LyoD has a long history of breaking changes _to the generated code_. The difference is that users need to explicitly regenerate.
2. We are aiming for this change to go into 5.0, breaking changes are fine (reasonably). @jadelkhoury make sure you write the ENV var name change to the changelog.
3. Let's not kid ourselves, it's easier to change a JVM property or override a Servlet Context parameter on the fly in an Application Server than change an ENV property. As Jad said, ENV properties are the main way of overriding configuration for Docker Containers (as long as it does not involve credentials etc.) and will be read on startup once. We are only making the override path consistent for all things in Lyo: the Servlet Context parameter is set in the web.xml, then an Application Server like Tomcat can apply Servlet Context parameter override, then that can be overridden with a JVM property and, finally, all of that can be overridden with an environment variable.

We had such overrides before but in various places the priority was different. We are simply straightening it out. I am working on a similar PR to the Lyo Core, we can scrutinize it a bit more then.

**Comment by berezovskyi** [comment]
For the posterity: I found only one instance of using `LYO_BASE` on my disk, for a k8s setup of a Lyo adaptor.

Service spec:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    run: server-rm
  name: server-rm
spec:
  # preset replica count conflicts with the HPA
  # replicas: 3
  selector:
    matchLabels:
      run: server-rm
  template:
    metadata:
      labels:
        run: server-rm
    spec:
      containers:
      - image: gitlab.example.com:1234/cnc19-group1/course-project/server-rm:latest
        imagePullPolicy: Always
        name: server-rm
        env:
        - name: K8S_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: LYO_BASE
          valueFrom:
            configMapKeyRef:
              name: oslc-config-v1
              key: url.rm
        ports:
        - containerPort: 8080
          protocol: TCP
        resources:
          limits:
            cpu: 500m
            memory: 384Mi
          requests:
            cpu: 500m
            memory: 384Mi
      imagePullSecrets:
      - name: regcred
      restartPolicy: Always
---
apiVersion: v1
kind: Service
metadata:
  labels:
    run: server-rm
  name: server-rm
spec:
  ports:
  - port: 8080
    protocol: TCP
  selector:
    run: server-rm
```

And then the `config.yaml` Config Map:

```
apiVersion: v1
kind: ConfigMap
metadata:
 name: oslc-config-v1
 namespace: default
data:
 url.rm: http://rm.cnative.berezovskyi.me:32149/
```

That "cloud native" URL is down, of course, but I hope you can appreciate why you don't want to rebuild a WAR file and a whole Docker image just to test it on a local k8s install or a remote one. This way, with flick of a finger, k8s ConfigMap can be updated on the fly.

**Comment by jadelkhoury** [comment]
so how do we set a ChangeLog (including the template Description when doing a pull request?

**Comment by berezovskyi** [comment]
@jadelkhoury like this? [`bab38f1` (#213)](https://github.com/eclipse/lyo.designer/pull/213/commits/bab38f1d0ea7baf355afc86cd375591dbc5cb364)

I added the changelog and a template to this PR, I am not sure if you really need a PR template as you are almost the sole committer here. But I can set it up it you want?

**Pull request merged**